src/dom.js

```diff
@@ -689,3 +690,40 @@ export function tryFocus(element) {
 export function isIframed(win) {
   return win.parent && win.parent != win;
 }
+
+/**
+ * Determines if this element is an AMP element
+ * @param {!Element} element
+ * @return {boolean}
+ */
+export function isAmpElement(element) {
+  const tag = element.tagName;
+  // Use prefix to recognize AMP element. This is necessary because stub
+  // may not be attached yet.
+  return startsWith(tag, 'AMP-') &&
+      // Some "amp-*" elements are not really AMP elements. :smh:
+      !(tag == 'AMP-STICKY-AD-TOP-PADDING' || tag == 'AMP-BODY');
+}
+
+/**
+ * Return a promise that resolve when an AMP element upgrade from HTMLElement
+ * to CustomElement
+ * @param {!Element} element
+ * @return {!Promise<!Element>}
+ */
+export function whenUpgradedToCustomElement(element) {
+  dev().assert(isAmpElement(element), 'element is not AmpElement');
+  if (element.createdCallback) {
```

Comment:
Not guaranteed. Adding it to DOM can happen much much later. So, I think, `createdCallback` is a good idea.